java/feathub-connectors/flink-connectors/flink-connector-prometheus/src/main/java/com/alibaba/feathub/flink/connectors/PrometheusSinkFunction.java

```diff
@@ -0,0 +1,351 @@
+        if (!labelsConfig.isEmpty()) {
+            Map<String, String> labelNames = new HashMap<>();
+            String[] kvs = labelsConfig.split(";");
+            for (String kv : kvs) {
+                int idx = kv.indexOf("=");
+                if (idx < 0) {
+                    throw new RuntimeException(
+                            String.format("Invalid prometheusPushGateway labels:%s", kv));
+                }
+
+                String labelName = kv.substring(0, idx);
+                String labelValue = kv.substring(idx + 1);
+                if (StringUtils.isNullOrWhitespaceOnly(labelName)) {
+                    throw new RuntimeException(
+                            String.format("Invalid labelName:%s must not be empty", labelName));
+                }
+                labelNames.put(labelName, labelValue);
+            }
+
+            return labelNames;
+        }
+
+        return Collections.emptyMap();
+    }
+
+    private static class MetricMeta implements Serializable {
+
+        private final String name;
+        private final List<String> labelNames;
+        private final List<String> labels;
+        private final int keyLabelCnt;
+
+        private final Map<String, Gauge.Child> children;
+
+        private transient Gauge.Child child;
+
+        private transient SimpleCollector<Gauge.Child> collector;
+
+        MetricMeta(
+                String name,
+                List<String> labelNames,
+                List<String> labels,
+                List<String> keyLabelNames) {
+            this.name = name;
+            this.labelNames = labelNames;
+            this.labels = labels;
+            this.children = new HashMap<>();
+            this.keyLabelCnt = keyLabelNames.size();
+            this.labelNames.addAll(keyLabelNames);
+        }
+
+        static MetricMeta fromColumn(Column column) {
+            Map<String, String> meta = parseLabels(column.getComment().orElse(""));
+            final List<String> nullLabelNames =
+                    meta.entrySet().stream()
+                            .filter(entry -> entry.getValue().equals("null"))
+                            .map(Map.Entry::getKey)
+                            .collect(Collectors.toList());
+
+            nullLabelNames.forEach(meta::remove);
+            return new MetricMeta(
+                    column.getName(),
+                    new ArrayList<>(meta.keySet()),
+                    new ArrayList<>(meta.values()),
+                    nullLabelNames);
+        }
+
+        void setCollector(SimpleCollector<Gauge.Child> collector) {
+            this.collector = collector;
+        }
+
+        Gauge.Child getChild(String keyLabel) {
+            return children.computeIfAbsent(
+                    keyLabel,
+                    label -> {
+                        final ArrayList<String> allLabels = new ArrayList<>(this.labels);
+                        for (int i = 0; i < keyLabelCnt; ++i) {
```

Comment:
Will `keyLabelCnt` always be `1` for map-typed value? If so, would it be better to make this requirement explicit in the code and in the document?